classes/battle/BattleManager.php

```diff
@@ -1054,6 +1114,116 @@ public function jutsuCollision(
             }
         }
 
+        // Barriers
+        if($player1->barrier && $player2_jutsu_is_attack) {
+            // Apply penalty against Genjutsu
+            if ($player2_jutsu->jutsu_type == Jutsu::TYPE_GENJUTSU) {
+                $player1->barrier *= self::GENJUTSU_BARRIER_PENALTY;
+            }
+
+            // Apply piercing
+            $player1->barrier *= (1 - $player2->piercing_percent);
```

Comment:
I feel weird about modifying the barrier value to signify that an attack is ignoring it, but don't have a better way to structure this on top of my head that reduces the proper amount, will follow up later